unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/BackpackEditorHUDController.cs

```diff
@@ -1,43 +1,210 @@
+            IBackpackEmotesSectionController backpackEmotesSectionController)
         {
             this.view = view;
-            this.view.Initialize();
             this.dataStore = dataStore;
+            this.userProfileBridge = userProfileBridge;
+            this.wearablesCatalogService = wearablesCatalogService;
+            this.backpackEmotesSectionController = backpackEmotesSectionController;
+
+            ownUserProfile.OnUpdate += LoadUserProfile;
             dataStore.HUDs.avatarEditorVisible.OnChange += SetVisibility;
             dataStore.HUDs.isAvatarEditorInitialized.Set(true);
             dataStore.exploreV2.configureBackpackInFullscreenMenu.OnChange += ConfigureBackpackInFullscreenMenuChanged;
             ConfigureBackpackInFullscreenMenuChanged(dataStore.exploreV2.configureBackpackInFullscreenMenu.Get(), null);
+            dataStore.exploreV2.isOpen.OnChange += ExploreV2IsOpenChanged;
+            backpackEmotesSectionController.OnNewEmoteAdded += OnNewEmoteAdded;
+            backpackEmotesSectionController.OnEmotePreviewed += OnEmotePreviewed;
+            backpackEmotesSectionController.OnEmoteEquipped += OnEmoteEquipped;
+            backpackEmotesSectionController.OnEmoteUnequipped += OnEmoteUnequipped;
             SetVisibility(dataStore.HUDs.avatarEditorVisible.Get());
         }
 
         public void Dispose()
         {
+            ownUserProfile.OnUpdate -= LoadUserProfile;
             dataStore.HUDs.avatarEditorVisible.OnChange -= SetVisibility;
             dataStore.exploreV2.configureBackpackInFullscreenMenu.OnChange -= ConfigureBackpackInFullscreenMenuChanged;
+            dataStore.exploreV2.isOpen.OnChange -= ExploreV2IsOpenChanged;
+
+            backpackEmotesSectionController.OnNewEmoteAdded -= OnNewEmoteAdded;
+            backpackEmotesSectionController.OnEmotePreviewed -= OnEmotePreviewed;
+            backpackEmotesSectionController.OnEmoteEquipped -= OnEmoteEquipped;
+            backpackEmotesSectionController.OnEmoteUnequipped -= OnEmoteUnequipped;
+            backpackEmotesSectionController.Dispose();
+
             view.Dispose();
         }
 
         public void SetVisibility(bool visible)
         {
             if (visible)
+            {
+                backpackEmotesSectionController.LoadEmotes();
+                LoadUserProfile(ownUserProfile, true);
                 view.Show();
+            }
             else
+            {
                 view.Hide();
+                view.ResetPreviewEmote();
+            }
+
+            SetRenderScale(visible);
         }
 
         private void SetVisibility(bool current, bool _) =>
             SetVisibility(current);
 
         private void ConfigureBackpackInFullscreenMenuChanged(Transform currentParentTransform, Transform previousParentTransform) =>
             view.SetAsFullScreenMenuMode(currentParentTransform);
+
+        private void ExploreV2IsOpenChanged(bool current, bool previous)
```

Comment:
its not completely clear to me why do we need to hook up with the explore v2

Reply:
Yeah, I've been analyzing that old flow and maybe it's not going to be needed to reference to explore v2 menu. Lets do a thing: lets remove it for now and we will see in the future if we need to apply some related logic.

Done: https://github.com/decentraland/unity-renderer/pull/4933/commits/d40b3b8e20c76108649695781774ab0023b6049e